docs/userguide/searchspace.md

```diff
@@ -1,46 +1,271 @@
+
+In BayBE, a search space is a union of two (potentially empty) subspaces. The [`SubspaceDiscrete`](baybe.searchspace.discrete.SubspaceDiscrete) contains all discrete parameters, i.e., all parameters which inherit from [`DiscreteParameter`](baybe.parameters.base.DiscreteParameter). The [`SubspaceContinuous`](baybe.searchspace.continuous.SubspaceContinuous) contains all continuous parameters, i.e., all parameters which inherit from [´ContinuousParameter´](baybe.parameters.base.ContinuousParameter).
+
+Depending on which of the subspaces are non-empty, a `SearchSpace` has exactly one of the three following [`SearchSpaceType`](baybe.searchspace.core.SearchSpaceType):
+1. If `SubspaceDiscrete` is non-empty but `SubspaceContinuous` is empty, the `SearchSpace` is *discrete* and has the type [`SearchSpaceType.DISCRETE`](baybe.searchspace.core.SearchSpaceType.DISCRETE).
+2. If `SubspaceDiscrete` is empty but `SubspaceContinuous` is non-empty, the `SearchSpace` is *continuous* and has the type [`SearchSpaceType.CONTINUOUS`](baybe.searchspace.core.SearchSpaceType.CONTINUOUS).
+3. If both subspaces are non-empty, the `SearchSpace` is *hybrid* and has the type [`SearchSpaceType.HYBRID`](baybe.searchspace.core.SearchSpaceType.HYBRID).
+
+## `SubspaceDiscrete`
+
+The `SubspaceDiscrete` contains all the discrete parameters of a `SearchSpace`. There are different ways of constructing this subspace.
+
+### Building from the Product of Parameter Values
+
+The method [`SearchSpace.from_product`](baybe.searchspace.discrete.SubspaceDiscrete.from_product) constructs the full cartesian product of the provided parameters:
+
+```python
+from baybe.parameters import NumericalDiscreteParameter
+from baybe.searchspace import SubspaceDiscrete
+
+parameters = [
+    NumericalDiscreteParameter(name="x0", values=[1, 2, 3]),
+    NumericalDiscreteParameter(name="x1", values=[4, 5, 6]),
+    NumericalDiscreteParameter(name="x2", values=[7, 8, 9]),
+]
+subspace = SubspaceDiscrete.from_product(parameters=parameters)
+```
+
+In this example, `subspace` has a total of 27 different parameter configuration.
+
+~~~
+       x0   x1   x2
+  0   1.0  4.0  7.0
+  1   1.0  4.0  8.0
+  2   1.0  4.0  9.0
+  ..  ...  ...  ...
+  24  3.0  6.0  7.0
+  25  3.0  6.0  8.0
+  26  3.0  6.0  9.0
+  
+  [27 rows x 3 columns]
+~~~
+
+### Constructing from a dataframe
+
+[`SubspaceDiscrete.from_dataframe`](baybe.searchspace.discrete.SubspaceDiscrete.from_dataframe) constructs a discrete subspace from a given dataframe.
+This method tries to infer the parameter types during the creation of the subspace.
+The parameter type is only necessary if special parameters should be used.
+
+```python
+df = pd.DataFrame(data=[[2, 5, 9], [3, 4, 7]], columns=["x0", "x1", "x2"])
+subspace = SubspaceDiscrete.from_dataframe(df)
+```
+
+In this case, it deduces that the parameters should be [`NumericalDiscreteParameter`](baybe.parameters.numerical.NumericalDiscreteParameter) and that no special encoding should be used.
```

Comment:
that sound too sophisticated, it doenst deduce anything

the statement about encoding should be removes as num params have no encoding

Reply:
Will remove that statement completely, it should be redundant with the now improved description at the beginning of the section.